File: play-kotlin-2.5.x/PlayKotlin/kmodules/K-Base/app/K/Controllers/ApiDoc/TemplateModel/ApiDefinition.java
package K.Controllers.ApiDoc.TemplateModel;

import K.Common.Helper;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;
import java.util.TreeSet;

/**
 * Created by kk on 15/1/12.
 */
public class ApiDefinition {
    public List<ApiGroup> groups;

    public ApiDefinition() {
        groups = new ArrayList<>();
    }

    public ApiGroup GetApiGroupByName(String group_name) {
        for (ApiGroup group : groups) {
            if (group.groupName.equals(group_name)) {
                return group;
            }
        }
        // 找不到 group, 则根据group_name, 创建一个新的
        ApiGroup group = new ApiGroup(group_name);
        groups.add(group);
        return group;
    }

    public Set<String> groupNames() {
        Set<String> names = new TreeSet<>();
        for (ApiGroup group : groups) {
            names.add(group.groupName);
        }
        return names;
    }

    public String ToMarkdownStr(String str) {
        return Helper.INSTANCE.EscapeMarkdown(str);
    }
}

File: play25kotlin/subProjects/ToolBox/app/k/common/apidoc/ApiDefinition.kt
package k.common.apidoc


import k.common.Helper

/**
 * Created by kk on 15/1/12.
 */
class ApiDefinition {

    var groups: MutableList<ApiGroup>

    init {
        groups = mutableListOf<ApiGroup>()

    }

    fun GetApiGroupByName(groupName: String): ApiGroup {

        var g = groups.find { it.groupName == groupName }

        if (g == null) {
            g = ApiGroup(groupName)
            groups.add(g)
        }

        return g
    }

    fun groupNames(): Set<String> {
        val names : MutableSet<String> = mutableSetOf()
        for (group in groups) {
            names.add(group.groupName)
        }
        return names
    }

    fun ToMarkdownStr(str: String): String {
        return Helper.EscapeMarkdown(str)
    }
}
